sql/core/src/main/scala/org/apache/spark/sql/jdbc/MySQLDialect.scala

```diff
@@ -41,6 +41,26 @@ private case object MySQLDialect extends JdbcDialect {
   override def quoteIdentifier(colName: String): String = {
     s"`$colName`"
   }
+  
+  override def parseTableName(tableName: String): String = {
+    val tableName1 = tableName.replace("\"", "").replace("\'", "")
+    if (tableName1.contains(".")) {
+      val tableNameList = tableName1.split('.')
+      tableNameList.foldLeft("") { (leftStr, rightStr) =>
```

Comment:
Woah, this is kinda dense and confusing. Can you please rewrite this in a simpler fashion and add comments? This is just my drive-by first impression.